flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/StateTable.java

```diff
@@ -199,6 +208,6 @@ public void setMetaInfo(RegisteredKeyValueStateBackendMetaInfo<N, S> metaInfo) {
 	@Nonnull
 	@Override
 	public StateSnapshotKeyGroupReader keyGroupReader(int readVersion) {
-		return StateTableByKeyGroupReaders.readerForVersion(this, readVersion);
+		return StateTableByKeyGroupReaders.readerForVersion(this, readVersion, keySerializer);
```

Comment:
Same here, seems redundant to pass in the key serializer since we're passing in the state table already. Maybe we should have `getKeySerializer` on `StateTable`.